code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -0,0 +1,350 @@
+	var/check_records = TRUE
+	var/arrest_type = FALSE
+	var/weaponscheck = TRUE
+	var/bikehorn = /obj/item/bikehorn
+	
+/obj/machinery/bot_core/honkbot
+	req_one_access = list(access_clown, access_robotics, access_mime)
+
+/mob/living/simple_animal/bot/honkbot/Initialize()
+	. = ..()
+	update_icon()
+	auto_patrol = TRUE
+	var/datum/job/clown/J = new/datum/job/clown
+	access_card.access += J.get_access()
+	prev_access = access_card.access
+
+/mob/living/simple_animal/bot/honkbot/proc/spam_flag_false() //used for addtimer
+	spam_flag = FALSE
+
+/mob/living/simple_animal/bot/honkbot/proc/sensor_blink()
+	icon_state = "honkbot-c"
+	addtimer(CALLBACK(src, .proc/update_icon), 5, TIMER_OVERRIDE|TIMER_UNIQUE)
+
+//honkbots react with sounds.
+/mob/living/simple_animal/bot/honkbot/proc/react_ping()
+	playsound(src, 'sound/machines/ping.ogg', 50, TRUE, -1) //the first sound upon creation!
+	spam_flag = TRUE
+	sensor_blink()
+	addtimer(CALLBACK(src, .proc/spam_flag_false), 18) // calibrates before starting the honk
+
+/mob/living/simple_animal/bot/honkbot/proc/react_buzz()
+	playsound(src, 'sound/machines/buzz-sigh.ogg', 50, TRUE, -1)
+	sensor_blink()
+
+/mob/living/simple_animal/bot/honkbot/bot_reset()
+	..()
+	target = null
+	oldtarget_name = null
+	anchored = FALSE
+	walk_to(src, 0)
+	last_found = world.time
+	spam_flag = FALSE
+
+/mob/living/simple_animal/bot/honkbot/set_custom_texts()
+
+	text_hack = "You overload [name]'s sound control system"
+	text_dehack = "You reboot [name] and restore the sound control system."
+	text_dehack_fail = "[name] refuses to accept your authority!"
+
+/mob/living/simple_animal/bot/honkbot/get_controls(mob/user)
+	var/dat
+	dat += hack(user)
+	dat += showpai(user)
+	dat += text({"
+	<TT><B>Honkomatic Bike Horn Unit v1.0.7 controls</B></TT><BR><BR>
+	Status: []<BR>
+	Behaviour controls are [locked ? "locked" : "unlocked"]<BR>
+	Maintenance panel is [open ? "opened" : "closed"]<BR>"},
+
+	"<A href='?src=[UID()];power=1'>[on ? "On" : "Off"]</A>")
+
+	if(!locked || issilicon(user) || user.can_admin_interact())
+		dat += "Auto Patrol <A href='?src=[UID()];operation=patrol'>[auto_patrol ? "On" : "Off"]</A><BR>"
+
+	return	dat
+
+/mob/living/simple_animal/bot/honkbot/proc/retaliate(mob/living/carbon/human/H)
+	threatlevel = H.assess_threat(src)
+	threatlevel += 6
+	if(threatlevel >= 4)
+		target = H
+		mode = BOT_HUNT
+
+/mob/living/simple_animal/bot/honkbot/attack_hand(mob/living/carbon/human/H)
+	if(H.a_intent == "harm")
+		retaliate(H)
+		addtimer(CALLBACK(src, .proc/react_buzz), 5)
+	return ..()
+
+// prank when trying to repair it
+/*mob/living/simple_animal/bot/honkbot/attackby(obj/item/W, mob/user, params)
+	if(W.tool_behaviour == TOOL_WELDER && user.a_intent != INTENT_HARM)
+		return
+	if(!W.tool_behaviour == TOOL_SCREWDRIVER && (W.force) && (!target) && (W.damtype != STAMINA) ) // Check for welding tool to fix #2432.
+		retaliate(user)
+		addtimer(CALLBACK(src, .proc/react_buzz), 5)
+	..()*/                          
```

Comment:
This chunk of code is still here commented out. Don't do that, just remove the whole comment block entirely unless it's going to be active.